src/utils/survey-event-receiver.ts

```diff
@@ -0,0 +1,60 @@
+import { Survey } from '../posthog-surveys-types'
+
+export class SurveyEventReceiver {
+    private eventRegistry: Map<string, string[]>
+    private surveysActivatedKey = 'surveysActivated'
+
+    constructor() {
+        this.eventRegistry = new Map<string, string[]>()
+    }
+
+    register(surveys: Survey[]): void {
+        surveys.forEach((survey) => {
+            if (survey.events && survey.events.length > 0) {
+                this.eventRegistry.set(survey.id, survey.events)
+            }
+        })
+    }
+
+    deRegister(survey: Survey): void {
+        if (this.eventRegistry.has(survey.id)) {
+            this.eventRegistry.delete(survey.id)
+        }
+    }
+
+    on(event: string): void {
+        const activatedSurveys: string[] = []
+        this.eventRegistry.forEach((events, surveyID) => {
+            if (events.includes(event)) {
+                activatedSurveys.push(surveyID)
+            }
+        })
+
+        let surveys: string[] = []
+        if (sessionStorage.getItem(this.surveysActivatedKey)) {
```

Comment:
hmm, I'm not sure if we should use the sessionStorage here, over posthog's persistence class -> https://github.com/PostHog/posthog-js/blob/main/src/posthog-persistence.ts#L55

Specially for cases where sessionStorage isn't supported, and this class has failover mechanisms.

The downside is we wouldn't know how long this data will survive (it can go in the cookie, localstorage, session storage, or memory, depending on what the user chooses) - but I think not a big deal, we can work around this, or change product requirements TM